events/reporter.go

```diff
@@ -511,3 +544,71 @@ func CloseEventReporter() {
 		reporter = nil
 	}
 }
+
+func newRing[T any](size int) *Ring[T] {
```

Comment:
container/ring (for this purpose list is the same) maintains structure using pointers. i want something that i can copy fast (e.g in this case i need to copy 100 8-byte pointers) and then use it without synchronized with original ring. for example i write events to stream, if i will hold lock during this time it may screw up consensus